File: prev/patterns/strobe.go
package patterns

import (
	"math"
	"time"

	"github.com/anderstorpsfestivalen/slisko/pkg/chassi"
	"github.com/anderstorpsfestivalen/slisko/pkg/utils"
)

type Strobe struct {
}

func (p *Strobe) Render(info RenderInfo, c *chassi.Chassi) {
	v := utils.Square(math.Sin(100 * time.Since(info.Start).Seconds()))
	for _, p := range c.LEDs {
		p.SetClamped(v, v, v)
	}

}

func (p *Strobe) Info() PatternInfo {
	return PatternInfo{
		Name:     "strobe",
		Category: "global",
	}
}

func (p *Strobe) Bootstrap(c *chassi.Chassi) {

}

File: engine/src/patterns/strobe.rs
use crate::chassi::Chassi;
use crate::pattern::{BootstrapCtx, Pattern, PatternInfo, RenderInfo};
use crate::utils;

/// Full-strand white strobe.
#[derive(Default)]
pub struct Strobe;

impl Pattern for Strobe {
    fn render(&mut self, info: &RenderInfo, c: &mut Chassi) {
        let value = utils::square(utils::sin_full(info.secs, 100.0));
        for pixel in &mut c.leds {
            pixel.set_clamped(value, value, value);
        }
    }

    fn info(&self) -> PatternInfo {
        PatternInfo {
            name: "strobe",
            category: "global",
        }
    }

    fn bootstrap(&mut self, _c: &Chassi, _ctx: &mut BootstrapCtx) {}
}
